conduit/plugins/importers/algod/mock_algod_test.go

```diff
@@ -7,48 +7,44 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/algorand/go-algorand-sdk/v2/client/v2/algod"
 	"github.com/algorand/go-algorand-sdk/v2/client/v2/common/models"
 	"github.com/algorand/go-algorand-sdk/v2/encoding/json"
 	"github.com/algorand/go-algorand-sdk/v2/encoding/msgpack"
 	"github.com/algorand/go-algorand-sdk/v2/types"
 )
 
+// algodCustomHandler is used by a silly muxer we created which brute forces the path and returns true if it handles the request.
+type algodCustomHandler = func(r *http.Request, w http.ResponseWriter) bool
+
 // AlgodHandler is used to handle http requests to a mock algod server
 type AlgodHandler struct {
-	responders []func(path string, w http.ResponseWriter) bool
+	responders []algodCustomHandler
 }
 
 // NewAlgodServer creates an httptest server with an algodHandler using the provided responders
-func NewAlgodServer(responders ...func(path string, w http.ResponseWriter) bool) *httptest.Server {
+func NewAlgodServer(responders ...algodCustomHandler) *httptest.Server {
 	return httptest.NewServer(&AlgodHandler{responders})
 }
 
 // NewAlgodHandler creates an AlgodHandler using the provided responders
-func NewAlgodHandler(responders ...func(path string, w http.ResponseWriter) bool) *AlgodHandler {
+func NewAlgodHandler(responders ...algodCustomHandler) *AlgodHandler {
 	return &AlgodHandler{responders}
 }
 
 // ServeHTTP implements the http.Handler interface for AlgodHandler
 func (handler *AlgodHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	for _, responder := range handler.responders {
-		if responder(req.URL.Path, w) {
+		if responder(req, w) {
```

Comment:
Switched interface to pass the entire request so that they can check the method. Ended up not using GetSyncRound, so it isn't actually used.